From 1e3f19821c862b9b552c72378a5b96f8d04b6431 Mon Sep 17 00:00:00 2001
From: Your Name <your.email@example.com>
Date: Mon, 10 Aug 2026 19:36:04 -0400
Subject: [PATCH] AudioFX: expose keepalive service for framework binding
 (MTR-025)

Mark AudioFxService exported with the MODIFY_AUDIO_SETTINGS permission so
system_server can bind the MusicFX keepalive component that the installed
package ships.

Change-Id: If400b247a661ec9fc9fc966f8a8e9815b60087bf
---
 AndroidManifest.xml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index cb10936..cc4e8a2 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -76,7 +76,10 @@
             </intent-filter>
         </activity>
 
-        <service android:name=".service.AudioFxService"/>
+        <service
+                android:name=".service.AudioFxService"
+                android:exported="true"
+                android:permission="android.permission.MODIFY_AUDIO_SETTINGS"/>
 
         <receiver android:name=".receiver.ServiceDispatcher"
             android:exported="true">
-- 
2.43.0

